### ISSUE TITLE
3.1 vs 3.2 HTTP server behavior change when `Host` header value not allowed

### ISSUE THREAD

**Issue by spoonincode** [AntelopeIO/leap#859]
On 3.1 when an HTTP client's `Host` header is not allowed by `http_plugin`, an empty 400 response is returned to the client. On 3.2+, the connection is simply closed without any headers sent to the client.

This is easy to reproduce by simply
```
nodeos --plugin eosio::chain_api_plugin
```
and then
```
curl -v http://localhost:8888/v1/chain/get_info
```
(`http_plugin` listens on 127.0.0.1 by default, and thus will not accept a `Host: localhost:8888` header)

IMO an error response should be sent (that is, the behavior change in 3.2 is undesirable). I would go further and suggest returning a small body describing the error would be very helpful.

**Comment by bhazzard** [comment]
Patched into 3.2, bring forward to 4.0 and beyond.

Caused confusion internally as well. Lack of information on why connection closes.

**Comment by ericpassmore** [comment]
Sorry mouse slip closing issue, and I shouldn’t have opened it. I ran into this issue last night. 

I feel http://local host should work. That’s a separate issue.
